pkg/common/telemetry/telemetry.go

```diff
@@ -0,0 +1,117 @@
+package telemetry
+
+import (
+	"io"
+	"time"
+
+	"github.com/armon/go-metrics"
+)
+
+type SinkConfig struct {
+	Logger      io.Writer
+	ServiceName string
+	StatsdAddr  string
+
+	StopChan <-chan struct{}
+}
+
+// sink carries a small bit of state used for cleanup/shutdown, and
+// implements the Sink interface.
+type sink struct {
+	m *metrics.Metrics
+
+	stopChan    <-chan struct{}
+	inmemCloser func()
+}
+
+// NewSink creates a new sink struct with the appropriate sinks configured.
+func NewSink(c *SinkConfig) Sink {
+	sinks := metrics.FanoutSink{}
+
+	// Always create an in-memory sink
+	interval := 1 * time.Second
+	retention := 1 * time.Hour
+	inmemSink := metrics.NewInmemSink(interval, retention)
+	sinks = append(sinks, inmemSink)
+
+	// Allow the in-memory sink to be signaled, printing stats to the log
+	inmemSignal := metrics.NewInmemSignal(inmemSink, metrics.DefaultSignal, c.Logger)
+
+	if c.StatsdAddr != "" {
```

Comment:
Perhaps change StatsdAddr to SinkAddr and include a SinkType. Have a select{ case c.SinkType:...} here instead.

Reply:
do you think there may be a use case for defining multiple sinks, i.e. statsd + prometheus? Also, not all sinks have an address - the prometheus sink may instead need a port number or something to bind the http server to.

alternatively, I could remove the StatsdAddr configurable altogether until we decide how we want it to work... inmem signal is enabled always

Reply:
allowing to configure any supported sink was the use case I was thinking of. We could certainly change it after we decide on a specific solution